### DISCUSSION TITLE
Is individual Takeout.zip processing supported?

### DISCUSSION THREAD

**Discussion by chippyhawkeye** [simulot/immich-go#171] [Q&A]
Hi,

Thanks for all the awesome work here!  I am trying to import around 1TB of Google photos.  I have 20 50GB zip archives.  I started by importing only the first which went very well except for many errors regarding "JSON File not found for this file".  I went ahead to the second takeout zip archive and it processed much quicker, but only added a small portion of the photos.  The summary is below, you can see it says there are 11171 files in the archive and it uploaded 342 files.  I'm curious if this is because it needs to process all the archives at once?  Any guidance is greatly appreciated!  -thanks  

Here is the summary:

Managing albums
Create the album Picture Party
Create the album St George Island
Update the album Family & friends
9 asset(s) added to the album "Family & friends"
Create the album CRG UGA
Scan of the sources:
 11171 files in the input
--------------------------------------------------------
  4564 photos
   330 videos
  6277 metadata files
   705 files with metadata
     0 discarded files
     0 files having a type not supported
     0 discarded files because in folder failed videos
 11171 input total (difference 0)
--------------------------------------------------------
   342 uploaded files on the server
     1 upgraded files on the server
   352 files already on the server
     0 discarded files because of options
     7 discarded files because duplicated in the input
     3 discarded files because server has a better image
     0 errors when uploading
   705 handled total (difference 4189)
Done.

**Comment by simulot** [comment]
No, it's not supported because of the way of takeout files  are clipped. Actual files are distributed accros archive parts.

You can either handle all zip files (use takeout*.zip) or you have to select only one year or album when doing the takeout procedure on the google web page.

**Reply by chippyhawkeye** [reply]
I'm nervous that it will fail with the full set at once.  Google says I'm using 1.1TB of space for my photos, but the takeout has 47 50GB files, 2.35TB!  Do you think it will be able to process all those in one go?  I guess it doesn't hurt to try...

**Reply by simulot** [reply]
Another caveat of google takeout is that images are duplicated between year's and album... 
Don't worry, immich-go detect them and upload only one instance of the image and assign it to the correct album.

You can run immich-go again and again... it will resume the work where it as been interrupted.

For a test, you can restrict the image to be upload using the -date option (https://github.com/simulot/immich-go#date-selection). Run immich-go with another date range to continue.
